onnxruntime/core/dlpack/dlpack_converter.cc

```diff
@@ -75,37 +75,6 @@ DLDataType GetDlpackDataType(const OrtValue& ort_value) {
   return dtype;
 }
 
-DLDevice GetDlpackDevice(const OrtValue& ort_value, const int64_t& device_id) {
-  ORT_ENFORCE(ort_value.IsTensor(), "Only OrtValues that are Tensors are currently supported");
-  DLDevice device;
-  device.device_id = static_cast<int>(device_id);
-  const Tensor& tensor = ort_value.Get<Tensor>();
-  const auto& location = tensor.Location();
-  switch (location.device.Type()) {
-    case OrtDevice::CPU:
-      device.device_type = DLDeviceType::kDLCPU;
-      break;
-    case OrtDevice::GPU:
-#ifdef USE_ROCM
-      device.device_type = DLDeviceType::kDLROCM;
-#else
-      device.device_type = DLDeviceType::kDLCUDA;
-#endif
-      break;
-    default:
-      ORT_THROW("Cannot pack tensors on this device.");
-  }
-
-  return device;
-}
-
-struct OrtDLManagedTensor {
```

Comment:
The new changed code doesn't use OrtDLManagedTensor or DlpackDeleter, maybe we don't need to move these two pieces. OrtDLManagedTensor doesn't need to add to header file.